src_MFsurfchem/MFsurfchem_functions.cpp

```diff
@@ -20,21 +21,24 @@ void InitializeMFSurfchemNamespace()
     // if n_ads_spec is set to 0 or not defined in the inputs file, quit the routine
     if (n_ads_spec==0) return;
 
-    surfcov0 = 0.;
+    std::vector<amrex::Real> surfcov0_tmp(MAX_SPECIES);
     // get initial surface coverage
-    pp.query("surfcov0",surfcov0);
+    pp.getarr("surfcov0",surfcov0_tmp,0,n_ads_spec);
```

Comment:
Note that "getarr" requires the user to supply this in the inputs file or the code will abort.  "queryarr" makes it optional.  I'm fine with any implementation, but if you switch to "queryarr" you need to load default values into surfcov0[]

See how prob_lo is treated in common_functions.cpp (default value, and use an "if" statement to load the parsed value into the GPUArray, since "queryarr" returns true if something is found)